### PR TITLE
Job description attribute SystemArchitecture

### PR DESCRIPTION
**Pull request by mtitov**
New attribute `SystemArchitecture` will include information about `CPUArchitecture`, `GPUArchitecture` and `SMT`, e.g. `{'cpu': '<cpu_arch>', 'gpu': '<gpu_arch>', 'smt': <smt_number>}`

### CODE REVIEW

**Comment by andre-merzky** [comment]
@mturilli : I suspect the default will vary per resource, not per adaptor, which makes this a wee bit more complicated...  One option would be to universally user the default `1` - that's easy to document :-D

**Comment by mturilli** [comment]
I see, maybe then always use the parameter for each resource so that it self-documents in the config file?

File: src/radical/saga/adaptors/slurm/slurm_job.py [hunk 1]
```diff
@@ -606,11 +609,10 @@ def _job_run(self, jd):
                 # gres resources are specified *per node*
                 assert(n_nodes), 'need unique number of cores per node'
 
-                if cpu_arch: gpu_arch = cpu_arch.lower()
-                else       : gpu_arch = 'p100'
-                if gpu_arch == 'k80': count = 4
-                else                : count = 2
-                script += "#SBATCH --gres=gpu:%s:%d\n" % (gpu_arch, count)  # Make sure we take a full GPU node
+                gpu_arch = 'p100' if not gpu_arch else gpu_arch.lower()
```
Comment (andre-merzky) [inline src/radical/saga/adaptors/slurm/slurm_job.py:612]: This is correct, but I don't think it makes the code clearer.

File: src/radical/saga/adaptors/slurm/slurm_job.py [hunk 2]
```diff
@@ -606,11 +609,10 @@ def _job_run(self, jd):
...
+                count = 4 if gpu_arch == 'k80' else 2
```
Comment (andre-merzky) [inline src/radical/saga/adaptors/slurm/slurm_job.py:613]: same

File: src/radical/saga/adaptors/slurm/slurm_job.py (hunk 1 above)
Comment (mtitov) [reply src/radical/saga/adaptors/slurm/slurm_job.py:612]: it is pretty standard and straightforward construction (reduces space and fulfill pylint/flake-requirements), but ok I'll return previous one back

Comment (andre-merzky) [reply src/radical/saga/adaptors/slurm/slurm_job.py:612]: *sigh* I know, thanks for bearing with me.

**Pull request merged**